https://www.hiv.lanl.gov/content/sequence/HIV/MAP/landmark.html
Ref: https://www.hiv.lanl.gov/components/sequence/HIV/asearch/query_one.comp?se_id=K03455

## MRC server:
/analyses2/data/Pangea_SJ/LANL_HXB2/

## SJ Laptop
/analyses2/data/Pangea_SJ/LANL_HXB2/

Datasets:

BHP
/Users/sanem/temp/GWAS/micgwas/HIV/BHP


LANL 
Query on 11th Feb 2023 (18,393)
	 - All complete genomes
	 - HIV 1
	 - All subtypes except N, O and P
	 - Including recombinants
	 
	 

grep ">" hiv-db.fasta | sed "s/>//" >seqNames.txt


+ Rename seqs with year at the end
+ Use /'s picking most meaningful info
	Country/name/subtype/year
+ Get subtype refs and generate subtype specific alignments
+ Sequence distribution
	- By continent region
	- By subtype


Phase 2:
+ Phylogenetics
+ Import / Export analysis
+ Phylogeography
+ Recombination
+ Selection pressure


## From CSV to fasta. Use the P to get a common pattern.
cat data/raw/pangea/seqs_pangea.csv | tr "," "\n" | sed 's/^P/>P/' | sed "s#P/##" > data/raw/pangea/seqs_pangea.fasta

cat raw/genebank/genebank_seqs_renamed.fasta <(echo) raw/lanl/lanl_seqs_renamed.fasta <(echo) raw/pangea/seqs_pangea.fasta > seqs_lanl_gb_pg.fasta

seqkit rmdup -n --ignore-case selected.fasta -o selected_uniq.fasta --dup-seqs-file selected_dups.fasta --dup-num-file selected_dups.txt

# 10468 duplicated records removed

seqkit rmdup -s --ignore-case selected_uniq.fasta -o selected_uniq_seqs.fasta --dup-seqs-file selected_dup_seqs.fasta --dup-num-file selected_dup_seqs.txt
# 2426 duplicated records removed

### split and align with aga - Commands for MRC server / adapt for mac if necessary
splitfasta selected_uniq_seqs_g.fasta

parallel 'aga --global {2} selected_uniq_seqs_g_{1}.fasta selected_uniq_seqs_g_{1}_aln.fasta' ::: $(seq 38624)  ::: "../../NC_001802.1.gb"

find ./*_aln.fasta | uniq | xargs -I{} sh -c "cat {}; echo ''" > aga_alignments.fasta

seqkit rmdup -n --ignore-case aga_alignments.fasta -o aga_alignment_uniq.fasta --dup-seqs-file aga_alignment_refs.fasta --dup-num-file aga_alignment_refs.txt

sed 's/\./N/g' aga_alignment_uniq.fasta | seqkit seq   -g -G "-" -o aga_alignment_uniq_v1.fasta


## or split and align
seqkit split seqs.fasta -s 3000 or -p 13
parallel 'mafft selected_uniq_seqs_g.part_{}.fasta > selected_uniq_seqs_g.part_{}_aln_mafft.fasta' ::: $(ls -1 *.fasta | cut -d "_" -f 5 | cut -d "." -f 1)

https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1002289
In this study we investigate how CTL escape mutations are likely to evolve in a vaccinated population and how CTL escape could affect the impact of an HIV vaccine. Specifically, we ask the following four questions:
 How would a CTL-inducing vaccine affect the progression of an HIV-epidemic?
 How would selection, reversion and transmission of CTL escape mutants affect the impact of a vaccine?
 Should a vaccine span a broad or narrow range of CTL epitopes?
 Should a vaccine target epitopes restricted by rare or common HLA alleles?
 
https://www.nature.com/articles/s41591-021-01590-5
Research priorities for an HIV cure: International AIDS Society Global Scientific Strategy 2021


https://retrovirology.biomedcentral.com/articles/10.1186/s12977-021-00565-1
Addressing an HIV cure in LMIC,  2021

https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0028889
The effective number of codons, GC content, and codon usage data corresponding to each gene of HIV-1 and SIVcpz were calculated using CodonW software (http://codonw.sourceforge.net).


https://www.frontiersin.org/articles/10.3389/fmicb.2017.01083/full
The evolutionary rates of the orthologous env genes representing all three types under analysis i.e., RP, SP, and LTNP (with reference to progressors) were calculated using Codeml program included in the PAML software package (ver. 4.5) (Nei and Gojobori, 1986; Yang, 2007) (http://abacus.gene.ucl.ac.uk/software/paml.html) with runmode = −2 and CodonFreq = 1.

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5447501/
Sequence-specific transcription factors (TFs) are key regulators of biological processes that function by binding to transcriptional regulatory regions (e.g., promoters, enhancers) to control the expression of their target genes.

Each TF typically recognizes a collection of similar DNA sequences, which can be represented as binding site motifs using models such as position weight matrices (PWMs)

https://www.hiv.lanl.gov/content/sequence/NEWALIGN/align.html
LANL alignments are generated by an iterative process between automated alignment using HMMER and manual editing using MASE, BioEdit, Se-Al or AliView. Most gaps have been introduced in multiples of 3 bases to maintain open reading frames. Any alignment is a compromise between optimal alignment, readability, and an attempt to keep codons intact. In particular the 'Other SIV' sequences are difficult to align, so please consider these as a rough starting point for your analyses.

Brian Foley: https://www.pangea-hiv.org/files/190719-brian-foley.pdf
Typical alignment of HIV-1 env by MAFFT or similar tools. Cannot be translated to amino acid sequences.

Alignment with muscle always ends in core dump:
Segmentation fault (core dumped)

MAFFT extremely slow. Run not complete after over a week.

https://ftp.ncbi.nlm.nih.gov/genomes/refseq/viral/Human_immunodeficiency_virus_1/reference/GCF_000864765.1_ViralProj15476/

/analyses2/data/Pangea_SJ/aga_hiv/selected_uniq_seqs_g_split_files/agaAlignments/agaAlignments
nextalign run -r LANL_HXB2/K03455.fasta selected_uniq_seqs_g.fasta \
 --output-all nextalign_out  \
 --input-gene-map LANL_HXB2/HBXB2_K03445_features.gff3 \
 --include-reference \
 --output-fasta selected_uniq_seqs_g_aln.fasta \
 --output-basename nextalign
 
 
 /analyses2/data/Pangea_SJ/aga_hiv/selected_uniq_seqs_g_split_files/agaAlignments/agaAlignments
nextalign run -r ../../../../refs/GCF_000864765.1_ViralProj15476_genomic.fna aga_alignment_uniq_v1.fasta \
 --output-all nextalign_out  \
 --input-gene-map ../../../../refs/GCF_000864765.1_ViralProj15476_genomic.gff \
 --include-reference \
 --output-fasta aga_alignment_uniq_v1_aln.fasta \
 --output-basename nextalign

find ./*_aln.fasta | uniq | xargs -I{} sh -c "cat {}; echo ''" > aga_alignments.fasta

seqkit grep -v -n -p 'NC_001802.1 Human immunodeficiency virus 1, complete genome' aga_alignments.fasta > aga_alignments_v1.fasta

mafft --thread 16 --threadtb 5 --threadit 0 --reorder --kimura 1 --large --retree 2 selected_uniq_seqs_g.fasta > ../align.selected_uniq_seqs_g_mafft.fasta  #based on mafft server https://mafft.cbrc.jp/
mafft --nomemsave --thread 20 --auto selected_uniq_seqs_g.fasta > selected_uniq_seqs_g_mafft.fasta  ## was faster


USA/viroverse: -> USA/viroverse.
## HIV sequence QC

FastTree -gtr -nt -boot 100 selected_uniq_seqs_g_mafft_trimmed_2pt.fasta  > selected_uniq_seqs_g_mafft_trimmed_2pt.fasttree.nwk

augur align [-h] --sequences FASTA [FASTA ...] [--output OUTPUT]
                   [--nthreads NTHREADS] [--method {mafft}]
                   [--reference-name NAME] [--reference-sequence PATH]
                   [--remove-reference] [--fill-gaps]
                   [--existing-alignment FASTA] [--debug]
                   
../../LANL_HXB2/HBXB2_K03445_features.gff3
../../LANL_HXB2/K03455.fasta

nextalign run -r ../LANL_HXB2/K03455.fasta selected_uniq_seqs_g_mafft_trimmed_2pt.fasta \
 --output-all nextalign_out  \
 --input-gene-map ../LANL_HXB2/HBXB2_K03445_features.gff3 \
 --include-reference \
 --output-fasta selected_uniq_seqs_g_mafft_trimmed_2pt_aln.fasta \
 --max-indel 1200
 --output-basename nextalign

/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences selected_uniq_seqs_g.fasta --output selected_uniq_seqs_g_augur_aln.fasta   --nthreads 24 --method mafft --reference-sequence ../LANL_HXB2/K03455.fasta  --remove-reference --fill-gaps --debug

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8160537/
#Runs:
mafft --reorder --anysymbol --nomemsave --adjustdirection --thread 24 augur_selected_uniq_seqs_g_aln.fasta.to_align.fasta 1> augur_selected_uniq_seqs_g_aln.fasta 2> augur_selected_uniq_seqs_g_aln.fasta.log

## If you supply a reference sequence to augur align, augur will include that reference sequence in the alignment and strip all insertions relative to that reference. This will guarantee a consistent reference coordinate system and genome annotation later on.

mafft --nomemsave --thread 20 --auto all_consensus_seqs_chrom2.fasta > all_consensus_seqs_chrom2_aln.fasta ##Anmol


FastTree -gtr -nt -boot 100 selected_uniq_seqs_g_augur_aln.fasta  > selected_uniq_seqs_g_augur_aln.fasttree.nwk

/analyses/software/miniconda3b/envs/phylogenetics/bin/augur tree --alignment selected_uniq_seqs_g_augur_aln.fasta --output selected_uniq_seqs_g_augur_aln.iq.nwk --nthreads 24
## Runs:
 iqtree2 -ntmax 24 -s selected_uniq_seqs_g_augur_aln-delim.fasta -m GTR -ninit 2 -n 2 -me 0.05 -nt AUTO -redo > selected_uniq_seqs_g_augur_aln-delim.iqtree.log

Recombination analysis

/usr/local/bin/hyphy gard -h

/usr/local/bin/hyphy gard alignment selected_uniq_seqs_g_mafft_trimmed_2pt.fasta type nucleotide 

## https://github.com/veg/hyphy/issues/1055 # the error is that there are too few sites for the number of sequences to run the inference. This is an inherent method limitation. You can prune some sequences out and try again with a smaller dataset.
## https://github.com/veg/hyphy/issues/980

https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0279597
Roadmap to the study of gene and protein phylogeny and evolution—A practical guide


# goalign clean seqs -i selected_uniq_seqs_g_mafft_trimmed_2pt_aln.fasta -c 0.5 -o selected_uniq_seqs_g_mafft_trimmed_2pt_aln_clean_ga.fasta -t 10

python /usr/local/biotools/tn93_cluster.py -i selected_uniq_seqs_g_augur_aln.fasta -o selected_uniq_seqs_g_augur_aln_tn93.fasta -j selected_uniq_seqs_g_augur_aln_tn93.json --threshold 0.3 -m 38625 -r ../LANL_HXB2/K03455.fasta

/usr/local/bin/hyphy gard alignment selected_uniq_seqs_g_augur_aln_tn93.fasta

>Loaded a nucleotide multiple sequence alignment with **19407** sequences, **9719** sites (9719 of which are variable) from `/analyses2/data/Pangea_SJ/aga_hiv/selected_uniq_seqs_g_augur_aln_tn93.fasta`
Minimum size of a partition is set to be 38811 sites
Error:
The alignment is too short to permit c-AIC based model comparison. Need at least 77629 sites for 19407 sequences to fit a two-partiton model. in call to assert(gard.minExpectedSites<=gard.numSites, error_msg);

## Centroids sequences by identity
usearch -cluster_fast selected_uniq_seqs_g_augur_aln.fasta -id 0.9 -centroids selected_uniq_seqs_g_augur_aln_centroids.fasta #Get the centroids

#Clusters by identity (for 85%, 90% and 95%)
vsearch -cluster_fast ../selected_uniq_seqs_g_augur_aln.fasta.gz -id 0.85 -clusters selected_uniq_seqs_g_augur_aln_cluster.fasta #Get the clusters
clusters_0.85 ## 15 clusters
clusters_0.9  ## 352 clusters
clusters_0.95	## 16784 clusters

for i in `ls *.fasta*`; do nSeqs=$(echo ${i} | grep ">" | wc -l); paste ${i} $nSeqs >> cluster_summary.txt; done
for i in `ls *.fasta*`; do seq=$(echo ${i}); nSeqs=$(grep ">" ${i} | wc -l); echo ${seq} ${nSeqs} >> cluster_summary_0.9.txt; done

https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0275623

https://www.bioinformatics.org/sms2/ident_sim.html ## Check sequence identities


Read the RDP paper;
- Why is there disagreement between methods
- REDP and Bootscan to have more agreement

Wormhole send to_send
wormhole recieve receive-code

## Darren:
I dont really know RAPR but jpHMM is ok for looking at inter-subtype recombinants if you can figure out how to use it properly.  
You may get some odd results when looking at subtype A and G recombinants.  jpHMM should give you highly accurate breakpoint 
locations for other recombinants though.

SAC decides on the scientific aspects of 
ETWG will comprise experts in EQA and can help with developing of the testing assesments which are reviewed by the SAC.
Africa CDC to be cordinate
NIMS-Network information management system



ls -1 *.fasta* > clusters_0.9.txt
for c in `cat clusters_0.9.txt`; do cluster=$(echo $c); clusterSize=$(echo $ | grep ">" | wc -l); 

seqkit sample --number 150 selected_uniq_seqs_g_augur_aln_cluster.fasta173 > down_sampled_0.9/selected_uniq_seqs_g_augur_aln_cluster.fasta173

for c in `cat toSample.txt`; do seqkit sample --number 150 ${c} > down_sampled_0.9/${c}; done

## for larger seq numbers seqkit does not return the desired number and it most subsets from the begining. 
## https://meme-suite.org/meme/doc/fasta-subsample.html //ports didn't work install vi mamba - https://anaconda.org/bioconda/meme
fasta-subsample selected_uniq_seqs_g_augur_aln_cluster.fasta173 150 > down_sampled_0.9/selected_uniq_seqs_g_augur_aln_cluster.fasta173

## under 10Ks
for c in `cat okay.txt`; do cp ${c} down_sampled_0.9/; done

## Copy and then downsample
for c in `cat toSample.txt`; do cp ${c} down_sampled_0.9/; done
for c in `cat toSample.txt`; do fasta-subsample ${c} 10000  > down_sampled_0.9/${c}; done


grep ">" selected_uniq_seqs_g_augur_aln_cluster.fasta173 | shuf |  head -n 150 | sed "s/>//" | gxargs -d '\n' -I {} seqkit grep -np {} selected_uniq_seqs_g_augur_aln_cluster.fasta173 > down_sampled_0.9/selected_uniq_seqs_g_augur_aln_cluster.fasta173


https://www.hiv.lanl.gov/content/sequence/NEWALIGN/help.html#RIP


RIP alignment. "/Users/sanem/Google\ Drive/PANGEA/LANL_HXB2/HIV1_RIP_2020_genome_DNA.fasta" -  Some of the sequences already in our alignment, so only adding the missing ones in subseted into HIV1_RIP_2020_genome_DNA_add.fasta


## Now strip of all gap characters: biostars.org/p/302104/
	#The following will not remove - from header lines, and remove all - from sequences:
	sed '/^>/! s/\-//g' test.fas
	#The following will not remove - from header lines, and remove - only from the end of lines:
	sed '/^>/! s/\-$//g' test.fas
	
cp HIV1_RIP_2020_genome_DNA_add.fasta HIV1_RIP_2020_genome_DNA_add_nogaps.fasta

sed '/^>/! s/\-//g' HIV1_RIP_2020_genome_DNA_add_nogaps.fasta > HIV1_RIP_2020_genome_DNA_add_nogaps_g.fasta # This will strip all dashes, then import to geneious to tidy up the sequences. Just save again from geneious and all the spaces will be gone.


scp HIV1_RIP_2020_genome_DNA_add_nogaps_g.fasta sanjames@mrcad.mrc.ac.za@krisp.mrc.ac.za:/analyses2/data/Pangea_SJ/aga_hiv/

## in /analyses2/data/Pangea_SJ/aga_hiv/aln
cat ../HIV1_RIP_2020_genome_DNA_add_nogaps_g.fasta <(echo) ../selected_uniq_seqs_g.fasta > selected_uniq_seqs_g_wRefs.fasta

/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences selected_uniq_seqs_g_wRefs.fasta  --output selected_uniq_seqs_g_augur_aln.fasta   --nthreads 24 --method mafft --reference-sequence ../LANL_HXB2/K03455.fasta  --remove-reference --fill-gaps --debug

# What is the length of all my sequences? Need the emboss package (install via conda)
infoseq -auto -only -name -length -noheading selected_uniq_seqs_g_wRefs.fasta

## Try split align merge
#/analyses2/data/Pangea_SJ/aga_hiv/aln/

seqkit split seqs.fasta -p 13

cd selected_uniq_seqs_g_wRefs.fasta.split

parallel -j 6 '/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences selected_uniq_seqs_g_wRefs.{1}.fasta  --output selected_uniq_seqs_g_augur_aln.{1}.fasta   --nthreads 6 --method mafft --reference-sequence {2}  --remove-reference --fill-gaps --debug' ::: $(ls *.part_* | cut -d "." -f 2) ::: "../../../LANL_HXB2/K03455.fasta"

touch selected_uniq_seqs_g_augur_aln_merged.fasta

for i in $(ls *_aln.part_*.fasta | cut -d "." -f 2 | uniq); do cat selected_uniq_seqs_g_augur_aln."${i}".fasta <(echo)  >> selected_uniq_seqs_g_augur_aln_merged.fasta; done

infoseq -auto -only -name -length -noheading  selected_uniq_seqs_g_augur_aln_merged.fasta | less

## Re-create tree
## vi find and replace all. https://linuxize.com/post/vim-find-replace/
:%s/:/_/
:%s/:/_/g [all]

FastTree -gtr -nt -boot 100 selected_uniq_seqs_g_augur_aln_merged.fasta  > selected_uniq_seqs_g_augur_aln.fasttree.nwk


/analyses/software/miniconda3b/envs/phylogenetics/bin/augur tree --alignment selected_uniq_seqs_g_augur_aln_merged.fasta --output selected_uniq_seqs_g_augur_aln.iq_tree.nwk --nthreads 24

#Test
aga --global ../../LANL_HXB2/HBXB2_K03455.gb 02_AG.NG.x.IBNG.L39106.fasta 02_AG.NG.x.IBNG.L39106_aga.fasta
scp sanjames@mrcad.mrc.ac.za@krisp.mrc.ac.za:/analyses2/data/Pangea_SJ/aga_hiv/aln/02_AG.NG.x.IBNG.L39106_aga.fasta .

## Reconstruct clusters
#in /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters
../selected_uniq_seqs_g_wRefs.fasta.split/selected_uniq_seqs_g_augur_aln_merged.fasta ##from alignment

sed '/^>/! s/N//g' cluster85wRefs.fasta14 > cluster85wRefs_noNs.fasta14
sed '/^>/! s/n//g' cluster85wRefs_noNs.fasta14 > cluster85wRefs_noNns.fasta14
# Extract reference set.
# Ids from ref refs_add.txt and refs_intheData.txt

seqkit grep -r -f HIV1_RIP_2020_genome_DNA_seqNames.txt ../selected_uniq_seqs_g_wRefs.fasta.split/selected_uniq_seqs_g_augur_aln_merged.fasta > HIV1_RIP_2020_genome_DNA.aln.fasta

# Generate clusters
# 85
vsearch -cluster_fast ../selected_uniq_seqs_g_wRefs.fasta.split/selected_uniq_seqs_g_augur_aln_merged.fasta -id 0.85 -clusters cluster85.fasta #Get the clusters

for c in $(../*.fasta* | xargs -n1 basename | cut -d "." -f2); do cat ../../HIV1_RIP_2020_genome_DNA.aln.fasta cluster85.${c} > cluster85wRefs.${c}
# 90
vsearch -cluster_fast ../selected_uniq_seqs_g_wRefs.fasta.split/selected_uniq_seqs_g_augur_aln_merged.fasta -id 0.9 -clusters cluster90.fasta #Get the clusters


## Unwrap lines / remove line breaks in fasta file
https://stackoverflow.com/questions/15857088/remove-line-breaks-in-a-fasta-file
seqtk seq -l 0 test_unwrap_in.fa > test_unwrap_out.fa


FastTree -gtr -nt -boot 100 cluster85wRefs.fasta  > cluster85wRefs_aln.fasttree.nwk

## cluster sequences by identity
python ~/Downloads/SDT_Mac64/SDT_Mac64.py /Users/sanem/Google Drive/PANGEA/data/clusters_0.85/wRefs/recluster/cluster85wRefs_unwrap.fasta muscle

## pangea refs
/analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/HIV1_RIP_2020_genome_DNA.aln.fasta
## Last seq in Refset: CentralAfricanRepublic/H.CF.1990.056.AF005496/1990
## sed  -i 's/^/ref_/' HIV1_RIP_2020_genome_DNA_refNames.txt     //append ref to the beginning of each reference sequence
 

## append refs
for c in $(ls ../*.fasta*  | xargs -n1 basename | cut -d "." -f2); do cat ../../../../HIV1_RIP_2020_genome_DNA.aln.fasta <(echo) ../cluster85.${c} > cluster85wRefs.${c}; done


### Try to align just the similar clusters...do I get a better alignment?
ref=/usr/local/biotools/shiver/info/B.FR.83.HXB2_LAI_IIIB_BRU.K03455.fasta

/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences selected_uniq_seqs_g_wRefs.fasta  --output selected_uniq_seqs_g_augur_aln.fasta   --nthreads 24 --method mafft --reference-sequence $ref  --remove-reference --fill-gaps --debug

# I ran this previously, 
# parallel -j 6 '/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences selected_uniq_seqs_g_wRefs.{1}.fasta  --output selected_uniq_seqs_g_augur_aln.{1}.fasta   --nthreads 6 --method mafft --reference-sequence {2}  --remove-reference --fill-gaps --debug' ::: $(ls *.part_* | cut -d "." -f 2) ::: "/analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta"

parallel -j 6 '/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs.{1}  --output cluster85wRefs_aln.{1}   --nthreads 24 --method mafft --reference-sequence {2}  --remove-reference --fill-gaps --debug' ::: $(ls *fasta* | cut -d "." -f 2) ::: "/analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta"

/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs.fasta00  --output cluster85wRefs_aln.fasta00   --nthreads 24 --method mafft --reference-sequence /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta  --remove-reference --fill-gaps --debug

### 
cluster85.fasta90*  -  34 clusters. Opted to drop and use 87 with 10 clusters

### 
The thinking here was that aligning more identical sequences would improve the alignments. So take the raw sequences, append the reference set to each sub-cluster and run the alignment to reduce the amount of diversity that needs to be dealt with.

/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs.fasta00  --output cluster85wRefs_aln.fasta00  --nthreads 24 --method mafft --reference-sequence /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta  --remove-reference --fill-gaps --debug ## if I trust to just align, the duplicates result in error, so I will first deduplicate.

seqkit rmdup --ignore-case cluster85wRefs.fasta00 -o cluster85wRefs_uniq.fasta00 --dup-seqs-file cluster85wRefs_dups.fasta00 --dup-num-file cluster85wRefs_dups00.txt


for i in `cat /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/rename_refs.txt`;  do  on=$(echo $i | cut -d "," -f1);  nn=$(echo $i | cut -d "," -f2);  /usr/bin/sed -i 's#$on#$nn#g' cluster85wRefs.fasta09;  done ## not running on the server, try this on my laptop

for i in `cat rename_refs.txt`;  do  on=$(echo $i | cut -d "," -f1);  nn=$(echo $i | cut -d "," -f2);  sed -i '' "s#$on#$nn#g" cluster85wRefs.fasta09;  done		## this works fine

## Drop dups:

seqkit rmdup --ignore-case cluster85wRefs.fasta00 -o cluster85wRefs_uniq.fasta00 --dup-seqs-file cluster85wRefs_dups.fasta00 --dup-num-file cluster85wRefs_dups00.txt
seqkit rmdup --ignore-case cluster85wRefs.fasta01 -o cluster85wRefs_uniq.fasta01 --dup-seqs-file cluster85wRefs_dups.fasta01 --dup-num-file cluster85wRefs_dups01.txt
seqkit rmdup --ignore-case cluster85wRefs.fasta02 -o cluster85wRefs_uniq.fasta02 --dup-seqs-file cluster85wRefs_dups.fasta02 --dup-num-file cluster85wRefs_dups02.txt
seqkit rmdup --ignore-case cluster85wRefs.fasta03 -o cluster85wRefs_uniq.fasta03 --dup-seqs-file cluster85wRefs_dups.fasta03 --dup-num-file cluster85wRefs_dups03.txt
seqkit rmdup --ignore-case cluster85wRefs.fasta04 -o cluster85wRefs_uniq.fasta04 --dup-seqs-file cluster85wRefs_dups.fasta04 --dup-num-file cluster85wRefs_dups04.txt
seqkit rmdup --ignore-case cluster85wRefs.fasta05 -o cluster85wRefs_uniq.fasta05 --dup-seqs-file cluster85wRefs_dups.fasta05 --dup-num-file cluster85wRefs_dups05.txt
seqkit rmdup --ignore-case cluster85wRefs.fasta06 -o cluster85wRefs_uniq.fasta06 --dup-seqs-file cluster85wRefs_dups.fasta06 --dup-num-file cluster85wRefs_dups06.txt
seqkit rmdup --ignore-case cluster85wRefs.fasta07 -o cluster85wRefs_uniq.fasta07 --dup-seqs-file cluster85wRefs_dups.fasta07 --dup-num-file cluster85wRefs_dups07.txt
seqkit rmdup --ignore-case cluster85wRefs.fasta08 -o cluster85wRefs_uniq.fasta08 --dup-seqs-file cluster85wRefs_dups.fasta08 --dup-num-file cluster85wRefs_dups08.txt
seqkit rmdup --ignore-case cluster85wRefs.fasta09 -o cluster85wRefs_uniq.fasta09 --dup-seqs-file cluster85wRefs_dups.fasta09 --dup-num-file cluster85wRefs_dups09.txt

/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs_uniq.fasta00  --output cluster85wRefs_uniq_aln.fasta00  --nthreads 24 --method mafft --reference-sequence /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta  --remove-reference --fill-gaps --debug
/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs_uniq.fasta01  --output cluster85wRefs_uniq_aln.fasta01  --nthreads 24 --method mafft --reference-sequence /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta  --remove-reference --fill-gaps --debug
/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs_uniq.fasta02  --output cluster85wRefs_uniq_aln.fasta02  --nthreads 24 --method mafft --reference-sequence /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta  --remove-reference --fill-gaps --debug
/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs_uniq.fasta03  --output cluster85wRefs_uniq_aln.fasta03  --nthreads 24 --method mafft --reference-sequence /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta  --remove-reference --fill-gaps --debug
/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs_uniq.fasta04  --output cluster85wRefs_uniq_aln.fasta04  --nthreads 24 --method mafft --reference-sequence /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta  --remove-reference --fill-gaps --debug
/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs_uniq.fasta05  --output cluster85wRefs_uniq_aln.fasta05  --nthreads 24 --method mafft --reference-sequence /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta  --remove-reference --fill-gaps --debug
/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs_uniq.fasta06  --output cluster85wRefs_uniq_aln.fasta06  --nthreads 24 --method mafft --reference-sequence /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta  --remove-reference --fill-gaps --debug
/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs_uniq.fasta07  --output cluster85wRefs_uniq_aln.fasta07  --nthreads 24 --method mafft --reference-sequence /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta  --remove-reference --fill-gaps --debug
/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs_uniq.fasta08  --output cluster85wRefs_uniq_aln.fasta08  --nthreads 24 --method mafft --reference-sequence /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta  --remove-reference --fill-gaps --debug

parallel -j 6 '/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs_{1}  --output cluster85wRefs_aln.{1}   --nthreads 24 --method mafft --reference-sequence {2}  --remove-reference --fill-gaps --debug' ::: $(ls *_uniq.fasta* | cut -d "_" -f 2) ::: "/analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta"

## break_cluster_1

vsearch -cluster_fast cluster85.fasta1 -id 87 -clusters cluster85.fasta1 #Get the clusters

mkdir wRefs && cd wRefs/

## add refseqs to each cluster
for c in $(ls ../*.fasta*  | xargs -n1 basename | cut -d "." -f2); do cat /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta <(echo) ../cluster85.${c} > cluster85wRefs.${c}; done

## remove dups
/usr/local/bin/parallel 'seqkit rmdup --ignore-case cluster85wRefs.{} -o cluster85wRefs_uniq.{} --dup-seqs-file cluster85wRefs_dups.{} --dup-num-file cluster85wRefs_dups.{}.txt' ::: $(ls *.fasta*  | xargs -n1 basename | cut -d "." -f2) # but no dups in this bunch so moving forward with the original sequences

#K03455|HIVHXB2CG refseq name

/usr/local/bin/parallel '/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs.{}  --output cluster85wRefs_aln.{} --nthreads 24 --method mafft --reference-name "K03455|HIVHXB2CG"   --remove-reference --fill-gaps --debug' ::: $(ls *.fasta*  | xargs -n1 basename | cut -d "." -f2)

#### start Derek-Abott.
/analyses2/data/derek-abbot-recombination

# Append refset
cat /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/HIV1_RIP_2020_genome_DNA.aln.fasta <(echo) run_consensus_v1.fasta > run_consensus_v1_wRefs.fasta

for i in `cat /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/rename_refs.txt`;  do  on=$(echo $i | cut -d "," -f1);  nn=$(echo $i | cut -d "," -f2);  /usr/bin/sed -i 's#$on#$nn#g' run_consensus_v1_wRefs.fasta;  done ## not running on the server, try this on my laptop

/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences run_consensus_v1_wRefs.fasta  --output run_consensus_v1_wRefs_aln.fasta   --nthreads 6 --method mafft --reference-sequence /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta  --remove-reference --fill-gaps --debug
#### end Derek-Abott.

Thur. June 29, 2023
## rename refs and re-align clusters
## alignments

cluster85 /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/cluster85/wRefs/deduped/msa ## all clusters with less than 8K seqs
break_cluster85_0 /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/cluster85/break_cluster85_0/87/wRefs/msa ## break alignment 0 further
break_cluster85_1 /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/cluster85/break_cluster85_1/wRefs/uniq/msa ## break alignment 1 futher

for c in $(ls ../*.fasta*  | xargs -n1 basename | cut -d "." -f2); do cat /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/HIV1_RIP_2020_genome_DNA.aln.fasta <(echo) ../cluster85.${c} > cluster85wRefs.${c}; done

for f in `ls *fasta*`; do
for i in `cat /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/rename_refs.txt`;  do  on=$(echo $i | cut -d "," -f1);  nn=$(echo $i | cut -d "," -f2);  /usr/bin/sed -i "s#$on#$nn#g" ${f};  done ## Server needs double quotes for sed
done

parallel -j 6 'seqkit rmdup --ignore-case ../cluster85wRefs.{} -o cluster85wRefs_uniq.{} --dup-seqs-file cluster85wRefs_dups.{} --dup-num-file cluster85wRefs_dups.{}.txt' ::: $(ls ../*.fasta* | xargs -n1 basename | cut -d "." -f 2)

parallel -j 6 '/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences ../cluster85wRefs_uniq.{1}  --output cluster85wRefs_uniq_aln.{1}   --nthreads 24 --method mafft --reference-sequence {2}  --remove-reference --fill-gaps --debug' ::: $(ls ../*_uniq.fasta* | xargs -n1 basename | cut -d "." -f 2) ::: "/analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta"

## break_cluster_00

vsearch -cluster_fast cluster85.fasta00 -id 0.9 -clusters cluster85.fasta00 #Get the clusters

mkdir wRefs && cd wRefs/

## add refseqs to each cluster
for c in $(ls ../*.fasta*  | xargs -n1 basename | cut -d "." -f2); do cat /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta <(echo) ../cluster85.${c} > cluster85wRefs.${c}; done

## remove dups from clusters - duplicates will make the alignment to fail...also a problem when creating phylogenies

mkdir -p deduped/dups

/usr/local/bin/parallel 'seqkit rmdup --ignore-case cluster85wRefs.{} -o deduped/cluster85wRefs_uniq.{} --dup-seqs-file deduped/dups/cluster85wRefs_dups.{} --dup-num-file deduped/dups/cluster85wRefs_dups.{}.txt' ::: $(ls *.fasta*  | xargs -n1 basename | cut -d "." -f2) # but no dups in this bunch so moving forward with the original sequences

#K03455|HIVHXB2CG refseq name
mkdir msa
/usr/local/bin/parallel '/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs_uniq.{}  --output msa/cluster85wRefs_uniq_aln.{} --nthreads 24 --method mafft --reference-name "K03455|HIVHXB2CG"   --remove-reference --fill-gaps --debug' ::: $(ls *.fasta*  | xargs -n1 basename | cut -d "." -f2)


## break_cluster_0/break_cluster_03 (89% identity)

vsearch -cluster_fast cluster85.fasta03 -id 0.89 -clusters cluster85.fasta03 #Get the clusters

mkdir wRefs && cd wRefs/

## add refseqs to each cluster
for c in $(ls ../*.fasta*  | xargs -n1 basename | cut -d "." -f2); do cat /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta <(echo) /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/HIV1_RIP_2020_genome_DNA.aln.fasta <(echo) ../cluster85.${c} > cluster85wRefs.${c}; done

for f in `ls *fasta*`; do
for i in `cat /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/rename_refs.txt`;  do  on=$(echo $i | cut -d "," -f1);  nn=$(echo $i | cut -d "," -f2);  /usr/bin/sed -i "s#$on#$nn#g" ${f};  done
done

## remove dups from clusters - duplicates will make the alignment to fail...also a problem when creating phylogenies
mkdir -p deduped/dups

/usr/local/bin/parallel 'seqkit rmdup --ignore-case cluster85wRefs.{} -o deduped/cluster85wRefs_uniq.{} --dup-seqs-file deduped/dups/cluster85wRefs_dups.{} --dup-num-file deduped/dups/cluster85wRefs_dups.{}.txt' ::: $(ls *.fasta*  | xargs -n1 basename | cut -d "." -f2) # but no dups in this bunch so moving forward with the original sequences

#K03455|HIVHXB2CG refseq name
mkdir msa
/usr/local/bin/parallel '/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs_uniq.{}  --output msa/cluster85wRefs_uniq_aln.{} --nthreads 24 --method mafft --reference-name "K03455|HIVHXB2CG"   --remove-reference --fill-gaps --debug' ::: $(ls *.fasta*  | xargs -n1 basename | cut -d "." -f2)

cd msa 
mv *.insertions.csv *.log *.pre_aligner.fasta *.post_aligner.fasta metadata/

## break_cluster_1/break_cluster_14 (89% identity)

vsearch -cluster_fast ../cluster85.fasta14 -id 0.89 -clusters cluster85.fasta14 #Get the clusters

mkdir wRefs && cd wRefs/

## add refseqs to each cluster
for c in $(ls ../*.fasta*  | xargs -n1 basename | cut -d "." -f2); do cat /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta <(echo) /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/HIV1_RIP_2020_genome_DNA.aln.fasta <(echo) ../cluster85.${c} > cluster85wRefs.${c}; done

for f in `ls *fasta*`; do
for i in `cat /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/rename_refs.txt`;  do  on=$(echo $i | cut -d "," -f1);  nn=$(echo $i | cut -d "," -f2);  /usr/bin/sed -i "s#$on#$nn#g" ${f};  done
done

for c in $(ls ../*.fasta*  | xargs -n1 basename | cut -d "." -f2); do cat /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta <(echo) cluster85s_uniq.${c} > cluster85wRefs_uniq.${c}; done

## remove dups from clusters - duplicates will make the alignment to fail...also a problem when creating phylogenies

for i in `cat /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/rename_refs.txt`;  do  on=$(echo $i | cut -d "," -f1);  nn=$(echo $i | cut -d "," -f2);  /usr/bin/sed -i 's#$on#$nn#g' run_consensus_v1_wRefs.fasta;  done ## not running on the server, try this on my laptop

cd msa 
mv *.insertions.csv *.log *.pre_aligner.fasta *.post_aligner.fasta metadata/


mkdir -p deduped/dups

/usr/local/bin/parallel 'seqkit rmdup --ignore-case cluster85wRefs.{} -o deduped/cluster85wRefs_uniq.{} --dup-seqs-file deduped/dups/cluster85wRefs_dups.{} --dup-num-file deduped/dups/cluster85wRefs_dups.{}.txt' ::: $(ls *.fasta*  | xargs -n1 basename | cut -d "." -f2) # but no dups in this bunch so moving forward with the original sequences

#K03455|HIVHXB2CG refseq name
mkdir msa
/usr/local/bin/parallel '/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs_uniq.{}  --output msa/cluster85wRefs_uniq_aln.{} --nthreads 24 --method mafft --reference-name "K03455|HIVHXB2CG"   --remove-reference --fill-gaps --debug' ::: $(ls *.fasta*  | xargs -n1 basename | cut -d "." -f2)


### Break cluster cluster85.fasta030 into two equal parts
cluster85.part_001.fasta030  
cluster85.part_002.fasta030

rename -- . _  *.fasta030 ## convert the first . to an _

seqkit split cluster85.fasta030 -p 2 mv these two into a new directory split_030

## add refseqs to each cluster
 #cat /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta <(echo) /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/HIV1_RIP_2020_genome_DNA.aln.fasta <(echo) ../cluster85_part_001.fasta030 > cluster85_part_001.fasta030
 #cat /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta <(echo) /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/HIV1_RIP_2020_genome_DNA.aln.fasta <(echo) ../cluster85.part_002.fasta030 > cluster85_part_002.fasta030

# On the new mac:
/Users/sanemj/Google Drive/My\ Drive/Pangea/Analysis/refs/K03455.fasta
/Users/sanemj/Google Drive/My\ Drive/Pangea/Analysis/refs/HIV1_RIP_2020_genome_DNA.aln.fasta

for f in `ls *fasta*`; do
for i in `cat /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/rename_refs.txt`;  do  on=$(echo $i | cut -d "," -f1);  nn=$(echo $i | cut -d "," -f2);  /usr/bin/sed -i "s#$on#$nn#g" ${f};  done
done

/usr/local/bin/parallel '' ::: $(ls *.fasta*  | xargs -n1 basename | cut -d "." -f2)

seqkit rmdup --ignore-case cluster85_part001wRefs.fasta030 -o deduped/cluster85_part001wRefs_uniq.fasta030 --dup-seqs-file deduped/dups/cluster85_part001wRefs_dups..fasta030 --dup-num-file deduped/dups/cluster85_part001wRefs_dups.fasta030.txt
seqkit rmdup --ignore-case cluster85_part002wRefs.fasta030 -o deduped/cluster85_part002wRefs_uniq.fasta030 --dup-seqs-file deduped/dups/cluster85_part002wRefs_dups..fasta030 --dup-num-file deduped/dups/cluster85_part002wRefs_dups.fasta030.txt

/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85_part001wRefs_uniq.fasta030  --output msa/cluster85_part001wRefs_uniq_aln.fasta030 --nthreads 24 --method mafft --reference-name "K03455|HIVHXB2CG"   --remove-reference --fill-gaps --debug
/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85_part002wRefs_uniq.fasta030  --output msa/cluster85_part002wRefs_uniq_aln.fasta030 --nthreads 24 --method mafft --reference-name "K03455|HIVHXB2CG"   --remove-reference --fill-gaps --debug

  
## Also break cluster 030 by country. Botswana and Uganda quite problematic, so analysing those separately. Potentially do the same with the rest of the analysis.

for c in $(ls ../*.fasta*  | xargs -n1 basename | cut -d "." -f2); do cat /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta <(echo) /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/HIV1_RIP_2020_genome_DNA.aln.fasta <(echo) ../cluster85.${c} > cluster85wRefs.${c}; done


for f in `ls *fasta*`; do
> for i in `cat /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/rename_refs.txt`;  do  on=$(echo $i | cut -d "," -f1);  nn=$(echo $i | cut -d "," -f2);  /usr/bin/sed -i "s#$on#$nn#g" ${f};  done
> done

/usr/local/bin/parallel '/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs_uniq.{}  --output msa/cluster85wRefs_uniq_aln.{} --nthreads 24 --method mafft --reference-name "K03455|HIVHXB2CG"   --remove-reference --fill-gaps --debug' ::: $(ls *.fasta*  | xargs -n1 basename | cut -d "." -f2)

## Put refs at the top of the alignment for easier analysis later in RDP. This is particularly important for generation breakpoint plots.
seqkit grep -rp "ref_" cluster85wRefs_aln.uniq.fasta03 > cluster85wRefs_aln.uniq.fasta03.a
seqkit grep -vrp "ref_" cluster85wRefs_aln.uniq.fasta03 > cluster85wRefs_aln.uniq.fasta03.b
cat cluster85wRefs_aln.uniq.fasta03.a cluster85wRefs_aln.uniq.fasta03.b > cluster85wRefs_aln.uniq_srt.fasta03
rm cluster85wRefs_aln.uniq.fasta03.a cluster85wRefs_aln.uniq.fasta03.b

#NEXUS


#NEXUS
begin trees;
	tree tree_1 = [&R] (('Botswana/074-A-106-1-6_w4_333.1789282683.MK458097/':0.844156,'Botswana/074-A-106-1-6_w4_333.1789282683.MK458097/_1089-1216':0.585824)[&label=36]:0.039142,'Botswana/074-A-106-1-6_w4_333.1789282683.MK458097/_1449-4530':0.883298);
end;


### Phylogegenetics

iqtree -s cluster85wRefs_uniq_aln.fasta11.rdp5.rrr_nr.fas -m GTR -b 100 -nt 4

#whole genos
iqtree2 -s cluster85wRefs_uniq_aln.fasta3.rdp5.nr_rrr.fas -bb 1000 -m GTR -pre cluster85wRefs_uniq_aln.fasta3.rdp5.nr_rrr.fas.uf
FastTree -nt -gtr -gamma < cluster85wRefs_uniq_aln.fasta3.rdp5.nr_rrr.fas > cluster85wRefs_uniq_aln.fasta3.rdp5.nr_rrr.fast.tree

iqtree2 -s cluster85wRefs_uniq_aln.fasta2.rdp5.nr_rrr.fas  -b 100 -m GTR -pre cluster85wRefs_uniq_aln.fasta2.rdp5.nr_rrr.fas
iqtree2 -s cluster85wRefs_uniq_aln.fasta3.rdp5.nr_rrr.fas  -b 100 -m GTR -pre cluster85wRefs_uniq_aln.fasta3.rdp5.nr_rrr.fas

#genes
iqtree2 -s cluster85wRefs_uniq_aln.fasta3.rdp5.nr_gag_rrr.fas -bb 1000 -m GTR -pre cluster85wRefs_uniq_aln.fasta3.rdp5.nr_gag_rrr.fas.uf
iqtree2 -s cluster85wRefs_uniq_aln.fasta3.rdp5.nr_pol_rrr.fas -bb 1000 -m GTR -pre cluster85wRefs_uniq_aln.fasta3.rdp5.nr_pol_rrr.fas.uf


FastTree -nt -gtr -gamma < cluster85wRefs_uniq_aln.fasta3.rdp5.nr_pol_rrr.fas > cluster85wRefs_uniq_aln.fasta3.rdp5.nr_pol_rrr.fast.tree

### prep seqs for hyphy
hyphy /Applications/biotools/hyphy-analyses/codon-msa/pre-msa.bf --input cluster85wRefs_uniq_aln.fasta3.rdp5.nr_gag_rrr.fas

mafft --nomemsave --thread 2 --auto cluster85wRefs_uniq_aln.fasta3.rdp5.nr_gag_rrr.fas_protein.fas > cluster85wRefs_uniq_aln.fasta3.rdp5.nr_gag_rrr.fas_protein.msa

hyphy /Applications/biotools/hyphy-analyses/codon-msa/post-msa.bf --protein-msa cluster85wRefs_uniq_aln.fasta3.rdp5.nr_gag_rrr.fas_protein.msa --nucleotide-sequences cluster85wRefs_uniq_aln.fasta3.rdp5.nr_gag_rrr.fas_nuc.fas --output cluster85wRefs_uniq_aln.fasta3.rdp5.nr_gag_rrr.fas

### hyph selection
hyphy fel --alignment cluster85wRefs_uniq_aln.fasta3.rdp5.nr_gag_rrr.fas --tree cluster85wRefs_uniq_aln.fasta3.rdp5.nr_gag_rrr.fas.uf.treefile

http://hyphy.org/tutorials/CL-prompt-tutorial/#:~:text=Preparing%20input%20data%20for%20HyPhy&text=Most%20standard%20alignment%20formats%20are,be%20a%20Newick%2Dformatted%20phylogeny.

#prune outliers from aligment and tree
seqkit grep -rvf cluster85wRefs_uniq_aln.fasta3.rdp5.nr_pol_rrr_na_drop.txt cluster85wRefs_uniq_aln.fasta3.rdp5.nr_pol_rrr_na.fas > cluster85wRefs_uniq_aln.fasta3.rdp5.nr_pol_rrr_na_nol.fas

/Users/sanemj/Temp/Bioinformatics/general-scripts/phylogenetics/drop_tips_cmd.R ccluster85wRefs_uniq_aln.fasta3.rdp5.nr_pol_rrr_na.fas.uf.treefile cluster85wRefs_uniq_aln.fasta3.rdp5.nr_pol_rrr_na_drop.txt cluster85wRefs_uniq_aln.fasta3.rdp5.nr_pol_rrr_na.fas.uf.pruned.treefile

hyphy fel --alignment cluster85wRefs_uniq_aln.fasta3.rdp5.nr_pol_rrr_na_nol.fas --tree cluster85wRefs_uniq_aln.fasta3.rdp5.nr_pol_rrr_na.fas.uf.pruned.treefile


### local combined alignment with HXB2 at the top
cat /Users/sanemj/Google\ Drive/My\ Drive/Pangea/Analysis/refs/K03455.fasta cluster85wRefs_uniq_aln_nr.fasta0 cluster85wRefs_uniq_aln_nr.fasta1 cluster85wRefs_uniq_aln_nr.fasta2 cluster85wRefs_uniq_aln_nr.fasta3 cluster85wRefs_uniq_aln_nr.fasta4 cluster85wRefs_uniq_aln_nr.fasta5 cluster85wRefs_uniq_aln_nr.fasta6 cluster85wRefs_uniq_aln_nr.fasta7 cluster85wRefs_uniq_aln_nr.fasta8 cluster85wRefs_uniq_aln_nr.fasta9 cluster85wRefs_uniq_aln_nr.fasta10 cluster85wRefs_uniq_aln_nr.fasta11 cluster85wRefs_uniq_aln_nr.fasta12 cluster85wRefs_uniq_aln_nr.fasta13 cluster85wRefs_uniq_aln_nr.fasta14 cluster85wRefs_uniq_aln_nr.fasta15 cluster85wRefs_uniq_aln_nr.fasta16 cluster85wRefs_uniq_aln_nr.fasta17 cluster85wRefs_uniq_aln_nr.fasta18 cluster85wRefs_uniq_aln_nr.fasta19 > cluster85_allseqs_wHXB2_uniq_aln.fasta

grep -c ">"  cluster85_allseqs_wHXB2_uniq_aln.fasta

## Get seqids
grep ">" cluster85wRefs_uniq_aln_nr.fasta10 | sed "s/>//" | > cluster85wRefs_uniq_aln_nr.fasta10.seq_ids.txt

##### Phylogenetics:
## Root
France/B.FR.1983.HXB2-LAI-IIIB-BRU.K03455/1983

cat cluster85wRefs_uniq_aln.fasta0/cluster85_0_aln_rrr_nr.rdp5.fas cluster85wRefs_uniq_aln.fasta1/cluster85_1_aln_rrr_nr.rdp5.fas cluster85wRefs_uniq_aln.fasta2/cluster85_2_aln_rrr_nr.rdp5.fas cluster85wRefs_uniq_aln.fasta3/cluster85_3_aln_rrr_nr.rdp5.fas cluster85wRefs_uniq_aln.fasta5/cluster85_5_aln_rrr_nr.rdp5.fas cluster85wRefs_uniq_aln.fasta7/cluster85_7_aln_rrr_nr.rdp5.fas cluster85wRefs_uniq_aln.fasta12/cluster85_12_aln_rrr_nr.rdp5.fas cluster85wRefs_uniq_aln.fasta13/cluster85_13_aln_rrr_nr.rdp5.fas cluster85wRefs_uniq_aln.fasta14/cluster85_14_aln_rrr_nr.rdp5.fas cluster85wRefs_uniq_aln.fasta17/cluster85_17_aln_rrr_nr.rdp5.fas K03455.fasta > cluster85_combined_aln_rrr_nr.rdp5.fas
FastTree -nt -gtr -gamma cluster85_combined_subset_aln_rrr_nr.rdp5.fas > cluster85_combined_subset_aln_rrr_nr.rdp5.fas.fast.tree

# sample 50 instead of 100. The -i simply ignores case
seqkit grep -rif cluster_subset_tip_ids_50.txt cluster85_combined_aln_rrr_nr.rdp5.fas > cluster85_combined_subset_aln_rrr_nr_s50.rdp5.fas
FastTree -nt -gtr -gamma cluster85_combined_subset_aln_rrr_nr_s50.rdp5.fas > cluster85_combined_subset_aln_rrr_nr_s50.rdp5.fas.fast.tree

## Tree without the extra long branch
seqkit grep -nvf extra_long_branch_seq_ids.txt cluster85_combined_subset_aln_rrr_nr_s50.rdp5.fas > cluster85_combined_subset_aln_rrr_nr_s50_stable.rdp5.fas
FastTree -nt -gtr -gamma cluster85_combined_subset_aln_rrr_nr_s50_stable.rdp5.fas > cluster85_combined_subset_aln_rrr_nr_s50_stable.rdp5.fas.fast.tree

FastTree -nt -gtr -gamma -boot 100 cluster85_combined_subset_aln_rrr_nr_s50_stable.rdp5.fas > cluster85_combined_subset_aln_rrr_nr_s50_stable.rdp5.fas.boot.fast.tree

treetime --aln cluster85_combined_subset_aln_rrr_nr_s50_stable.rdp5.fas --tree cluster85_combined_subset_aln_rrr_nr_s50_stable.rdp5.fas.boot.fast.tree --dates metadata.tsv --clock-rate 0.0008 --reroot oldest --clock-std-dev 0.0004 > log.txt

cluster85_0_aln_rrr_nr_root_neseqs.rdp5.fas #neseqs - 'is no empty sequences'



//WhatsApp number +256772043790 - Perez//

### Cluster ML trees

## Cluster 0
FastTree -nt -gtr -gamma -boot 100 cluster85_0_aln_rrr_nr_root_neseqs.rdp5.fas > cluster85_0_aln_rrr_nr_root_neseqs.rdp5.fas.boot.fast.tree


## Cluster 1
FastTree -nt -gtr -gamma -boot 100 cluster85_1_aln_rrr_nr_root_neseqs.rdp5.fas > cluster85_1_aln_rrr_nr_root_neseqs.rdp5.fas.boot.fast.tree
iqtree2 --boot-trees  -T 12 -s cluster85_1_aln_rrr_nr_root_neseqs.rdp5.fas -bb 1000 -m MFP -pre cluster85_1_aln_rrr_nr_root_neseqs.rdp5.fas.uf

## Cluster 2
# Done on my laptop in IQT

## Cluster 3 - 
# Done on my laptop in IQtree

## cluster5
iqtree2 -s cluster85_5_aln_rrr_nr_root.rdp5.fas -bb 1000 -m GTR -pre cluster85_5_aln_rrr_nr_root.rdp5.fas.uf
FastTree -nt -gtr -gamma -boot 100 cluster85_5_aln_rrr_nr_root.rdp5.fas > cluster85_5_aln_rrr_nr_root.rdp5.fas.boot.fast.tree


## cluster7
iqtree2 -s cluster85_7_aln_rrr_nr_root.rdp5.fas -bb 1000 -m GTR -pre cluster85_7_aln_rrr_nr_root.rdp5.fas.uf
FastTree -nt -gtr -gamma -boot 100 cluster85_7_aln_rrr_nr_root.rdp5.fas > cluster85_7_aln_rrr_nr_root.rdp5.fas.boot.fast.tree


## Cluster 12
FastTree -nt -gtr -gamma -boot 100 cluster85_12_aln_rrr_nr_root_neseqs.rdp5.fas > cluster85_12_aln_rrr_nr_root_neseqs.rdp5.fas.boot.fast.tree
FastTree -nt -gtr -gamma -boot 100 cluster85_12_aln_rrr_nr_root_neseqs_pruned1.rdp5.fas > cluster85_12_aln_rrr_nr_root_neseqs_pruned1.rdp5.fas.boot.fast.tree  ## Two seqs removed


## Cluster 13
FastTree -nt -gtr -gamma -boot 100 cluster85_13_aln_rrr_nr_root_neseqs.rdp5.fas > cluster85_13_aln_rrr_nr_root_neseqs.rdp5.fas.boot.fast.tree
FastTree -nt -gtr -gamma -boot 100 cluster85_13_aln_rrr_nr_root_neseqs_pruned1.rdp5.fas > cluster85_13_aln_rrr_nr_root_neseqs_pruned1.rdp5.fas.boot.fast.tree

iqtree2  --boot-trees  -T 12 -s cluster85_13_aln_rrr_nr_root_neseqs.rdp5.fas -bb 1000 -m MFP -pre cluster85_13_aln_rrr_nr_root_neseqs.rdp5.fas.uf

## Cluster 14
iqtree2 -s  cluster85_14_aln_rrr_nr_root.rdp5.fas -bb 1000 -m GTR -pre  cluster85_14_aln_rrr_nr_root.rdp5.fas.uf


## Cluster 17
iqtree2 -s  cluster85_17_aln_rrr_nr_root.rdp5.fas -bb 1000 -m GTR -pre  cluster85_17_aln_rrr_nr_root.rdp5.fas.uf


#####
https://www.census.gov/data-tools/demo/hiv/#/map?s_datacode=R



####### difff:
cat /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta <(echo) /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/HIV1_RIP_2020_genome_DNA.aln.fasta <(echo) ../cluster85.fasta0 > cluster85wRefs.fasta0
cat /analyses2/data/Pangea_SJ/LANL_HXB2/K03455.fasta <(echo) /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/HIV1_RIP_2020_genome_DNA.aln.fasta <(echo) ../cluster85.fasta1 > cluster85wRefs.fasta1

for i in `cat /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/rename_refs.txt`;  do  on=$(echo $i | cut -d "," -f1);  nn=$(echo $i | cut -d "," -f2);  /usr/bin/sed -i "s#$on#$nn#g"  cluster85wRefs.fasta0;  done
for i in `cat /analyses2/data/Pangea_SJ/aga_hiv/aln/clusters/rename_refs.txt`;  do  on=$(echo $i | cut -d "," -f1);  nn=$(echo $i | cut -d "," -f2);  /usr/bin/sed -i "s#$on#$nn#g"  cluster85wRefs.fasta1;  done

seqkit rmdup --ignore-case cluster85wRefs.fasta0 -o deduped/cluster85wRefs_uniq.fasta0 --dup-seqs-file deduped/dups/cluster85wRefs_dups.fasta0 --dup-num-file deduped/dups/cluster85wRefs_dups.fasta0.txt
seqkit rmdup --ignore-case cluster85wRefs.fasta1 -o deduped/cluster85wRefs_uniq.fasta1 --dup-seqs-file deduped/dups/cluster85wRefs_dups.fasta1 --dup-num-file deduped/dups/cluster85wRefs_dups.fasta1.txt


/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs_uniq.fasta0  --output msa/cluster85wRefs_uniq_uniq_aln.fasta0 --nthreads 24 --method mafft --reference-name "K03455|HIVHXB2CG"   --remove-reference --fill-gaps --debug
/analyses/software/miniconda3b/envs/phylogenetics/bin/augur align --sequences cluster85wRefs_uniq.fasta1  --output msa/cluster85wRefs_uniq_uniq_aln.fasta1 --nthreads 24 --method mafft --reference-name "K03455|HIVHXB2CG"   --remove-reference --fill-gaps --debug



seqkit grep -rp "ref_" cluster85wRefs_uniq_aln.fasta13 > cluster85wRefs_uniq_aln.fasta13.a
seqkit grep -vrp "ref_" cluster85wRefs_uniq_aln.fasta13 > cluster85wRefs_uniq_aln.fasta13.b
cat cluster85wRefs_uniq_aln.fasta13.a cluster85wRefs_uniq_aln.fasta13.b > cluster85wRefs_aln.uniq_srt.fasta13
rm cluster85wRefs_uniq_aln.fasta13.a cluster85wRefs_uniq_aln.fasta13.b



 cluster85wRefs_aln.uniq_srt.fasta0
 Trim:
 first 496
 last 9223-9108
 
cluster85wRefs_aln.uniq_srt.fasta1
first 447
last 9285-9177

ATGGGTGCGAGAGCGTCAGTATTAAGAGGCGAAAAATTAGATGAATGGGAAAAAATTAGGCTAAGGCCAGGGGGAAAGAAACGTTATATGCTAAAACACATAATATGGGCAAGCAGGGAGCTGGAAAGATTTGCACTTAACCCTGGCCTTTTGGAGACATCAGAAGGCTGTAAACAGATAATGTCACAGCTACAACCAGCTCTTAAGACAGGAACAGAGGAACTTAAATCATTATACAACACAGTAGCAACTCTCTATTGTGTACATAGAAGGATAAAAGTACAAGACACCAAGGAACTTAGACAAGATAGAGGAAGAACAAAAACAATGAAAAACACAGCAGGCAGCGGCGGTGGAACCAGTCAAAATTAT
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------


tn93-cluster -f -o /usr/local/biotools/RASCL/results/B.1.1.7/ORF7a.query.json -t 0.0005 /usr/local/biotools/RASCL/results/B.1.1.7/ORF7a.query.m
sa.SA
usage: tn93-cluster [-h] [-o OUTPUT] [-a AMBIGS] [-l OVERLAP] [-t THERSHOLD] [-c CLUSTER-TYPE] [-m OUTPUT_MODE] [-g FRACTION] [-q] [FASTA]
tn93-cluster: error: unknown argument: -f
Command exection failed code 256
Reference seq_name



### Nextstrain:
rule all:
    input:
        auspice_json = "auspice/zika.json",

input_fasta = "data/sequences.fasta",
input_metadata = "data/metadata.tsv",
dropped_strains = "config/dropped_strains.txt",
reference = "config/zika_outgroup.gb",
colors = "config/colors.tsv",
lat_longs = "config/lat_longs.tsv",
auspice_config = "config/auspice_config.json"


mv auspice/zika.json auspice/hiv.json

2848 sequences:

augur index \
  --sequences data/sequences.fasta \
  --output results/sequence_index.tsv
  
augur filter \
  --sequences data/sequences.fasta \
  --sequence-index results/sequence_index.tsv \
  --metadata data/metadata.tsv \
  --include config/include.txt \
  --output results/filtered.fasta \
  --group-by country year month \
  --sequences-per-group 20 \
  --min-date 1984

### Try to skip
augur align \
  --sequences results/filtered.fasta \
  --reference-sequence config/K03455.gb \
  --output results/aligned.fasta \
  --fill-gaps

## Jump to tree
augur tree \
  --alignment results/filtered.fasta \
  --method fasttree \
  --output results/tree_raw.nwk
  
## FastTreeDblMP -nt -nosupport results/filtered.fasta 1> results/tree_raw.nwk 2> results/tree_raw.nwk.log
  
augur refine \
  --tree results/tree_raw.nwk \
  --alignment results/aligned.fasta \
  --metadata data/metadata.tsv \
  --output-tree results/tree.nwk \
  --output-node-data results/branch_lengths.json \
  --stochastic-resolve \
  --timetree \
  --coalescent opt \
  --date-confidence \
  --date-inference marginal \
  --clock-filter-iqd 4
  
augur traits \
  --tree results/tree.nwk \
  --metadata data/metadata.tsv \
  --output-node-data results/traits.json \
  --columns cluster_id Country \
  --confidence
  
augur ancestral \
  --tree results/tree.nwk \
  --alignment results/aligned.fasta \
  --output-node-data results/nt_muts.json \
  --inference joint
  
augur translate \
  --tree results/tree.nwk \
  --ancestral-sequences results/nt_muts.json \
  --reference-sequence config/K03455.gb \
  --output-node-data results/aa_muts.json
  
augur export v2 \
  --tree results/tree.nwk \
  --metadata data/metadata.tsv \
  --node-data results/branch_lengths.json \
              results/traits.json \
              results/nt_muts.json \
              results/aa_muts.json \
  --colors config/colors.tsv \
  --lat-longs config/lat_longs.tsv \
  --auspice-config config/auspice_config.json \
  --output auspice/hiv_tn93_s1.json
  
  ## New tree
  
  # sed '/^>/! s/\^//g' <seq2.fa >seq3.fa
  
  sed '/^>/ s/:/_/g' < selected_uniq_seqs_g_augur_aln.fasta  > selected_uniq_seqs_g_augur_aln_R1.fasta #remove : colons from sequence names.
  sed '/^>/ s/\'/_/g' < selected_uniq_seqs_g_augur_aln_R1.fasta > selected_uniq_seqs_g_augur_aln.fasta #remove 's colons from sequence names.
  sed '/^>/ s/--/_/g' < selected_uniq_seqs_g_augur_aln_R1.fasta  > selected_uniq_seqs_g_augur_aln.fasta
  
  FastTree -nt -gtr -gamma -boot 100 selected_uniq_seqs_g_augur_aln_R1.fasta > selected_uniq_seqs_g_augur_aln_R1.fast.tree
  
  sed 's/--/_/g' < selected_uniq_seqs_g_augur_aln_R1.fast.nwk  > selected_uniq_seqs_g_augur_aln.fast.nwk

  
  ## I could not get sed to replace the ', so I used it within vim.
  s/--/_/g for a single line or %s/'/_/g for the whole document.
  
  ## Also drop # (pound) signs from the Japanese sequences
  
  ## File name in Newick

  bootstrap=0.7
  geneticDistances=10
  maxClusterSize=8000
  
  Japan/#117.186940871.AB428551/2000  #11 more with #
  
  java –jar /Applications/biotools/ClusterPicker/ClusterPicker_1.2.5.jar selected_uniq_seqs_g_augur_aln_R1.fasta selected_uniq_seqs_g_augur_aln_R1_bf.fast.nwk $bootstrap $bootstrap $geneticDistance $maxClusterSize

seqkit rmdup --ignore-case refs_aln.fasta -o refs_aln_uniq.fasta --dup-seqs-file refs_aln_dups.fasta --dup-num-file refs_aln_dups.txt

seqkit grep -rf d_seq_names.txt selected_uniq_seqs_g_augur_aln_R1.fasta > d_seqs.fasta
cat refs_aln_uniq.fasta d_seqs.fasta > d_wRefs.fasta

for i in `cat rename.txt`;  do  on=$(echo $i | cut -d "," -f1);  nn=$(echo $i | cut -d "," -f2);  sed -i '' "s#$on#$nn#g" d_wRefs.fasta;  done


FastTree -nt -gtr -gamma -boot 100 d_wRefs.fasta > d_wRefs.boot.fast.tree


### Untrimmed tree
FastTree -nt -gtr -gamma -boot 100 selected_uniq_seqs_gg_mafft.fasta > selected_uniq_seqs_g_mafft.boot.fast.tree

virulign ../NC_001802.1-mod3.fasta ~/Google\ Drive/My\ Drive/Pangea/Analysis/sequences.fasta --exportAlphabet Nucleotides --exportReferenceSequence yes --exportWithInsertions yes --exportKind GlobalAlignment --progress yes > sequences.virulign.fasta 2> error.file
virulign ../NC_001802.1-coding.fasta ~/Google\ Drive/My\ Drive/Pangea/Analysis/sequences.fasta --exportAlphabet Nucleotides --exportReferenceSequence yes --exportWithInsertions yes --exportKind GlobalAlignment --progress yes > sequences.coding.virulign.fasta 2> error-coding.file
 